.github/workflows/sonarcloud.yml

```diff
@@ -9,16 +9,16 @@ on:
   pull_request:
 
 jobs:
-  sonarcloud:
-    name: SonarCloud
-    runs-on: ${{ matrix.os }}
-    if: "!contains(github.event.head_commit.message, '[skip ci]')"
-    strategy:
-      matrix:
-        os: [ubuntu-20.04]
+  build:
+    name: Build and analyze
+    runs-on: ubuntu-latest
+    env:
+      BUILD_WRAPPER_OUT_DIR: build_wrapper_output_directory # Directory where build-wrapper output will be placed
+      - name: Install sonar-scanner and build-wrapper
+        uses: SonarSource/sonarcloud-github-c-cpp@v1
 
     env:
-      SONAR_SCANNER_VERSION: 4.7.0.2747 # Find the latest version in the "Linux" link on this page:
+      SONAR_SCANNER_VERSION: 5.0.1.3006 # Find the latest version in the "Linux" link on this page:
```

Comment:
Do we need this ? Does the action not set it for us ?

Also, please keep `if: "!contains(github.event.head_commit.message, '[skip ci]')"`, it's a simple way to skip the CI for example when adding minor stuff such as comments.